lib/charms/data_platform_libs/v0/data_interfaces.py

```diff
@@ -2842,6 +2852,24 @@ def _on_relation_created_event(self, event: RelationCreatedEvent) -> None:
 
     def _on_relation_changed_event(self, event: RelationChangedEvent) -> None:
         """Event emitted when the database relation has changed."""
+        is_subordinate = False
+        remote_unit_data = None
+        for key in event.relation.data.keys():
+            if isinstance(key, Unit) and key.name.startswith(self.charm.app.name):
```

Comment:
Fix to get the remote unit data:
```suggestion
            if isinstance(key, Unit) and not key.name.startswith(self.charm.app.name):
```